Sources/NIOCrashTester/CrashTests+EventLoop.swift

```diff
@@ -0,0 +1,65 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import Foundation
+
+fileprivate let group = MultiThreadedEventLoopGroup(numberOfThreads: 2)
+
+struct EventLoopCrashTests {
+    let testMultiThreadedELGCrashesOnZeroThreads = CrashTest(
+        regex: "^Precondition failed: numberOfThreads must be positive"
+    ) {
+        try? MultiThreadedEventLoopGroup(numberOfThreads: 0).syncShutdownGracefully()
+    }
+
+    let testWaitCrashesWhenOnEL = CrashTest(
+        regex: #"^Precondition failed: BUG DETECTED: wait\(\) must not be called when on an EventLoop"#
+    ) {
+        let promise = group.next().makePromise(of: Void.self)
+        try? group.next().submit {
+            try? promise.futureResult.wait()
+        }.wait()
+    }
+
+    let testAssertInEventLoop = CrashTest(
+        regex: "^Precondition failed: file .*CrashTests\\+EventLoop.swift, line \(#line + 2)"
```

Comment:
Super minor, this feels slightly brittle setting the expected line like this, would it be better to just manually override the line in `assertInEventLoop` etc.?

Reply:
good idea!

Reply:
done